### PR TITLE
Create tests for metaedge_to_adjacency_matrix

### PR DESCRIPTION
**Pull request by zietzm**
Refs https://github.com/greenelab/hetmech/issues/36

Test defines a modified version of the original metaedge_to_adjacency matrix function that creates a numpy array and then converts it to a sparse matrix. It then checks whether matrix.py has a function that gives an identical sparse matrix. This test will pass if the new function version in matrix.py gives exactly a sparse matrix version of what the function originally gave.

### CODE REVIEW

File: hetmech/test_matrix.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    https://doi.org/10.1371/journal.pcbi.1004259.g002
+    '''
+
+    url = 'https://github.com/dhimmel/hetio/raw/{}/{}'.format(
+        '9dc747b8fc4e23ef3437829ffde4d047f2e1bdde',
+        'test/data/disease-gene-example-graph.json',
+    )
+    graph = hetio.readwrite.read_graph(url)
+
+    def inef_metaedge_to_adjacency_matrix(graph, metaedge, dtype=numpy.bool_):
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: You want to remove this complex logic from the test. For this initial PR, let's just try to add tests for the existing `metaedge_to_adjacency_matrix` function without adding any new functionality. In other words, you'll manually construct the adjecency matrices corresponding to the example graph. You'll hardcode these matrices.

Note that tests are different than other types of coding. In most coding, you want to use logic and avoid hardcoding. In tests, you want to hardcode the expected result and check that the function output matches.

File: hetmech/test_matrix.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+'''
+
+url = 'https://github.com/dhimmel/hetio/raw/{}/{}'.format(
+    '9dc747b8fc4e23ef3437829ffde4d047f2e1bdde',
+    'test/data/disease-gene-example-graph.json',
+)
+graph = hetio.readwrite.read_graph(url)
+
+
+def test_sparse_matrix():
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: Let's define a single test `test_metaedge_to_adjacency_matrix` that tests both the `GiG` and `GaD` matrixes are correct.

File: hetmech/test_matrix.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    mat = metaedge_to_adjacency_matrix(graph, 'GiG')
+    assert all([mat[2][0, 4], mat[2][0, 2], mat[2][6, 2], mat[2][2, 0]])
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: This is on the right track: it checks for the boolean value of certain cells. Let's be even more explicit by testing that the matrix object is exactly equal. You'll want to construct an actual test matrix like [here](https://github.com/greenelab/hetmech/blob/df8a1fca2193f375e8c8c79be76ac8ab255699ea/hetmech/test_diffusion.py#L14-L19).

File: hetmech/test_matrix.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+data as in test_degree_weight.py Figure 2D of Himmelstein &
+Baranzini (2015) PLOS Comp Bio.
+https://doi.org/10.1371/journal.pcbi.1004259.g002
+'''
+
+url = 'https://github.com/dhimmel/hetio/raw/{}/{}'.format(
+    '9dc747b8fc4e23ef3437829ffde4d047f2e1bdde',
+    'test/data/disease-gene-example-graph.json',
+)
+graph = hetio.readwrite.read_graph(url)
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: This should go inside of the test function (or in another function that the test calls). As is, this graph will be loaded whenever the module is imported.

File: hetmech/test_matrix.py [hunk 5]
```diff
@@ -1,28 +1,36 @@
...
+    Baranzini (2015) PLOS Comp Bio.
+    https://doi.org/10.1371/journal.pcbi.1004259.g002
+    """
+    url = 'https://github.com/dhimmel/hetio/raw/{}/{}'.format(
+        '9dc747b8fc4e23ef3437829ffde4d047f2e1bdde',
+        'test/data/disease-gene-example-graph.json',
+    )
+    graph = hetio.readwrite.read_graph(url)
 
+    mat_gig = metaedge_to_adjacency_matrix(graph, 'GiG')
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: I would do something like the following:

```python
row_names, column_names, adjacency_matrix = metaedge_to_adjacency_matrix(graph, 'GiG')
```

And let's also test the row/column names.

**Comment by dhimmel** [comment]
Let's also test Disease--localization--Tissue in both orderings: e.g. `DlT` and `TlD`. It will be good to test both orientations. These matrices should be small (2 x 2)

File: hetmech/test_matrix.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import hetio.readwrite
+from .matrix import metaedge_to_adjacency_matrix
+import numpy as np
+
+
+def test_metaedge_to_adjacency_matrix():
+    """
+    Test the functionality of metaedge_to_adjacency_matrix in
+    generating sparse matrices vs numpy arrays. Uses same test
```
Comment (dhimmel) [inline hetmech/test_matrix.py]: Let's remove "sparse matrices vs". Then ready to merge.

**Comment by dhimmel** [comment]
@zietzm I didn't realize you had write access. Oops. In the future, I'll merge unless I note otherwise. The main thing here is that I'd like to squash all the commits. Will do that on master now.

**Comment by zietzm** [comment]
Sorry about that! Will do next time.

**Comment by dhimmel** [comment]
@cgreene currently any Greene Lab member gets write on all repos. While potentially convenient, this scope is a bit broad IMO. See for example above. I think it's best to have write access explicitly granted. For example, I don't think I should be able to go around changing other lab members repos that I'm not involved in.

From this repos settings:

> The Greene Laboratory organization has their default repository permission set to write. This means that every member of this organization has write access to this repository, regardless of the team and collaborator access specified below.

**Comment by dhimmel** [comment]
@zietzm I squashed into https://github.com/greenelab/hetmech/commit/b11b5f699c879345261187e4b5a283109d88680a.

**Comment by kkloste** [comment]
Finally had a chance to look over some of this work -- looking good, @zietzm !
This will also make it easier to code sparse-matrix functionality and testing.

**Pull request merged**